### PR TITLE
resize image when shape is passed

### LINKED ISSUE #3093: Image is resized when using tool 'sketch'
### Describe the bug

I have an image with shape (height=80,width=848), and I want to use the 'sketch' tool to generate mask for it.
However, whether or not I use the gr.Image().style(height = 80, width = 848) or set shape parameters gr.Image(shape=(848,80)), the image is resized automatically which is not my desire.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

without .style
```
import gradio as gr
import os


def image_mod(input_image):
    init_image = input_image["image"]
    init_mask = input_image["mask"]
    return init_mask

with gr.Blocks() as demo:
    with gr.Column():
        i_image = gr.Image(tool='sketch',shape=(848,80))# .style(height = 80, width = 848)
        o_image = gr.Image(shape=(848,80))# .style(height = 80, width = 848)
        button = gr.Button()

    button.click(image_mod,inputs=i_image,outputs=o_image)


if __name__ == "__main__":
    demo.launch()
```

### Screenshot

without .style:
image

with .style:
image
image
The image even cover the button and other components.

### Logs

```shell
$ python sketch.py 

Thanks for being a Gradio user! If you have questions or feedback, please join our Discord server and chat with us: https://discord.gg/feTf9x3ZSB
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
```

### System Info

```shell
gradio version: 3.11.0
system: windows 11
python version: 3.9.7
```

### Severity

annoying

### PR DESCRIPTION
**Pull request by pngwn**
# Description

That was annoying.

Images uploaded when shape is present now get zoom/cropped. I think the behaviour is consistent with the server.

Fixes #3093

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by pngwn** [comment]
To test:

```py
import gradio as gr

gr.Interface(
    lambda x:(x["image"], x["mask"]),
    inputs=gr.Image(tool="sketch", shape=(100, 100)),
    outputs=["image", "image"],
).launch()
```

https://github.com/gradio-app/gradio/issues/3093#issuecomment-1442219338 has some good test cases too.

File: ui/packages/image/src/Sketch.svelte [hunk 1]
```diff
@@ -122,7 +158,15 @@
...
 				} else {
-					ctx.temp.drawImage(value_img, 0, 0);
+					const image = get_image_details();
+
+					ctx.temp.drawImage(
+						value_img,
+						image.x,
+						image.y,
+						image.width,
+						image.height
```
Comment (abidlabs) [inline ui/packages/image/src/Sketch.svelte:168]: Looks like there's some code repetition, perhaps refactor out to a separate method?

**Comment by abidlabs** [comment]
Hey @pngwn I don't think the way the frontend and backend are handling the cropping is actually consistent. If you try a simple example like this:

```py
import gradio as gr

gr.Interface(
    lambda x:(x["image"], x["mask"]),
    inputs=gr.Image(tool="sketch", shape=(100, 100)),
    outputs=["image", "image"],
).launch()
```

you'll notice that there is some discrepancy:

image


I think what happens is that the backend _first_ zooms and then crops if necessary (in order to get as much of the image as much possible), whereas the frontend _first_ crops and then zooms (that's why most of the cheetah image is gone in the frontend)

**Comment by pngwn** [comment]
Hmm. It should be zooming then cropping. Let me try again.

**Comment by pngwn** [comment]
Turns out I'm an idiot, square images weren't being handled. Also refactored the draw image into a function.

File: ui/packages/image/src/Sketch.svelte [hunk 2]
```diff
@@ -52,7 +52,14 @@
...
 				} else {
-					ctx.temp.drawImage(value_img, 0, 0);
+					const image = draw_cropped_image();
+					ctx.temp.drawImage(
+						value_img,
+						image.x,
+						image.y,
+						image.width,
+						image.height
+					);
```
Comment (abidlabs) [inline ui/packages/image/src/Sketch.svelte:62]: Might be a silly question, but do we still need this? I thought the last line of `draw_cropped_image()` does this

**Comment by abidlabs** [comment]
Very interesting. Works great now, LGTM @pngwn!

**Review by abidlabs — APPROVED** [review]

File: ui/packages/image/src/Sketch.svelte (hunk 2 above)
Comment (pngwn) [reply ui/packages/image/src/Sketch.svelte:62]: wait, did i leave one in?

Comment (pngwn) [reply ui/packages/image/src/Sketch.svelte:62]: i'll check

Comment (pngwn) [reply ui/packages/image/src/Sketch.svelte]: okay i did, it is gone now.
